### PR TITLE
Fe/feature/ri 5390 enhance rdi routing

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

File: redisinsight/ui/src/pages/rdi/instance/InstancePage.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    lastPageRef.current = getPageName(rdiInstanceId, pathname)
+  }, [pathname])
+
+  useEffect(() => {
+    if (isShouldChildrenRerender) setIsShouldChildrenRerender(false)
+  }, [isShouldChildrenRerender])
+
+  useEffect(() => {
+
+  }, [])
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/InstancePage.tsx:81]: do we need this useEffect?

File: redisinsight/ui/src/pages/rdi/instance-list/RdiInstancesListWrapper.spec.tsx [hunk 2]
```diff
@@ -117,6 +117,14 @@ describe('RdiInstancesListWrapper', () => {
...
             data: mockInstances
           }
+        },
+        app: {
+          ...state.app,
+          context: {
+            ...state.app.context,
+            lastPage: '',
+            contextRdiInstanceId: '',
+          }
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance-list/RdiInstancesListWrapper.spec.tsx:127]: please, mock only exact selector, not useSelector

File: redisinsight/ui/src/pages/rdi/pipeline-management/PipelineManagementPageRouter.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import React from 'react'
 import { Switch } from 'react-router-dom'
 import RouteWithSubRoutes from 'uiSrc/utils/routerWithSubRoutes'
+import { IRoute } from 'uiSrc/constants'
 
 export interface Props {
-  routes: any[];
+  routes: IRoute[];
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/pipeline-management/PipelineManagementPageRouter.tsx:7]: please, remove semicolon

File: redisinsight/ui/src/pages/rdi/instance/InstancePage.tsx (hunk 1 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/instance/InstancePage.tsx:81]: no, thank you

File: redisinsight/ui/src/pages/rdi/instance-list/RdiInstancesListWrapper.spec.tsx (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/instance-list/RdiInstancesListWrapper.spec.tsx:127]: Got it.

**Review by romansergeenkosofteq — APPROVED** [review]

**Pull request merged**